src/ProjectTracker.ts

```diff
@@ -0,0 +1,61 @@
+import * as _path from 'path';
+import * as fs from 'fs';
+
+class ProjectTracker {
+  private _last_weak_project: string;
+  private _last_strong_project: string;
+  private _last_directory: string;
+
+  constructor() {
+    this._last_weak_project = null;
+    this._last_strong_project = null;
+  }
+
+  findProject(path: string, recurseTimes: number = 10): string {
```

Comment:
be declarative about nullables
```suggestion
  findProject(path: string|null, recurseTimes: number = 10): string|null {
```